nomos-da/network/core/src/protocols/sampling/behaviour.rs

```diff
@@ -334,20 +336,40 @@ where
         message: sampling::SampleRequest,
         subnetwork_id: SubnetworkId,
         blob_id: BlobId,
-    ) -> Result<StreamHandlerFutureSuccess, SamplingError> {
-        let into_sampling_error = |error: std::io::Error| -> SamplingError {
-            SamplingError::Io {
-                peer_id: stream.peer_id,
-                error,
+    ) -> Result<StreamWriterFutureSuccess, StreamWriterFutureError> {
+        if let Err(error) = pack_to_writer(&message, &mut stream.stream).await {
+            return Err((
+                SamplingError::Io {
+                    peer_id: stream.peer_id,
+                    error,
+                },
+                Some(stream),
+            ));
+        }
```

Comment:
you didn't use `map_err`+`?` because of ownership issues with `stream`?

Reply:
Yes, `map_err` moves values, couldn't use.